### PR TITLE
Add namedpipereceiver

### PR DESCRIPTION
**Pull request by sinkingpoint**
**Description:**

This adds a new receiver that reads from a named pipe, using the previously created namedpipeinput stanza operator (#28841). Because that operator does the majority of the work, this is mostly boilerplate + a few tests to confirm that it works e2e.

Link to tracking Issue: #27234

**Testing:**

Added a few tests, loading the config, and actually creating a receiver to verify that it can read logs from the pipe.

**Documentation:**

None, yet.

/cc @djaglowski - where should I add docs for this if any?

### CODE REVIEW

**Comment by djaglowski** [comment]
> where should I add docs for this if any?

Certainly we need a readme for the component (i.e. `receiver/namedpipereceiver/README.md`).

Ideally we would also have a separate doc in `pkg/stanza/docs` but we can just capture this as an issue. It's not ideal that we have so much duplication between the two, but it's the pattern we currently. Once we have a readme for the receiver, it should be easy to tweak it for the operator.

File: receiver/namedpipereceiver/metadata.yaml
Comment (djaglowski) [inline receiver/namedpipereceiver/metadata.yaml:7]:
```suggestion
  distributions: []
```

**Review by djaglowski — COMMENTED** [review]
Code looks good, just need to include a readme and all the boilerplate updates like `.github` `versions.yaml`, etc. (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/28818 is a recent reference)

**Comment by sinkingpoint** [comment]
@djaglowski I've added those + a basic README. Tests seem to be passing as well

**Comment by djaglowski** [comment]
With https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29472, we no longer have a dependabot.yml file. @codeboten, is there an equivalent change we need to make for renovate?

File: receiver/namedpipereceiver/metadata.yaml
Comment (atoulme) [inline receiver/namedpipereceiver/metadata.yaml:6]: is this really in alpha?

File: receiver/namedpipereceiver/metadata.yaml
Comment (atoulme) [inline receiver/namedpipereceiver/metadata.yaml:7]: this is not in the contrib distribution yet.

**Comment by atoulme** [comment]
This PR makes @djaglowski the only codeowner of the component - @djaglowski please approve the PR, this seems a bit out of pace with how we describe adding components in CONTRIBUTING.md. @sinkingpoint I see in the original issue you have yourself as codeowner, is that still valid?

File: receiver/namedpipereceiver/metadata.yaml
Comment (djaglowski) [reply receiver/namedpipereceiver/metadata.yaml:6]: It will be once the PR is merged

File: receiver/namedpipereceiver/metadata.yaml
Comment (djaglowski) [reply receiver/namedpipereceiver/metadata.yaml:7]: I can be added once this PR is merged

**Comment by sinkingpoint** [comment]
Apologies - missed that there was something for me to do here. Will fix this up today

**Review by djaglowski — APPROVED** [review]

**Comment by djaglowski** [comment]
@sinkingpoint, I was just updating some things relating to code ownership and noticed that we still aren't able to assign you as an owner to this component because you're not a member of OpenTelemetry yet. Are you willing to [apply](https://github.com/open-telemetry/community/blob/main/community-membership.md#requirements)? I'm happy to be one of your sponsors.

**Comment by sinkingpoint** [comment]
Ah! You did mention. Sure - application PR here: https://github.com/open-telemetry/community/issues/1975

**Pull request merged**
